### ISSUE TITLE
Wrong exit code in nested sends invalid params

### ISSUE THREAD

**Issue by austinabell** [filecoin-project/test-vectors#93]
inside of `test-vectors/corpus/nested/nested_sends--fail-mismatch-params.json` and `test-vectors/corpus/nested/nested_sends--fail-missing-params.json` the exit code returned in the return value is `SysErrSenderInvalid` and should probably be `ErrSerialization` based on details of the test. (same issue as the marked invalid tests `fails-unparsable-constructor-params-via-init-actor`and `fails-unparsable-init-actor-exec-msg`)

**Comment by raulk** [comment]
@alanshaw is tracking the exit codes test vectors. There was a change needed in Lotus to make it comply in the manner you mention, @austinabell. Can't remember what the status of this is.

**Comment by austinabell** [comment]
> @alanshaw is tracking the exit codes test vectors. There was a change needed in Lotus to make it comply in the manner you mention, @austinabell. Can't remember what the status of this is.

Awesome, yeah that was mentioned. I was just opening issue for tracking and because I was asked to the other day

**Comment by alanshaw** [comment]
I'm just revisiting this as I noticed https://github.com/filecoin-project/lotus/pull/3184 was merged.

I've submitted a PR (https://github.com/filecoin-project/test-vectors/pull/133) to verify that the exit code in `fail-mismatch-params` and `fail-missing-params` is now `ErrSerialization` 🎉.

However, https://github.com/filecoin-project/lotus/pull/3184 does not fix this issue for [`fails-unparsable-constructor-params-via-init-actor`](https://github.com/filecoin-project/test-vectors/blob/7d3becbeb5b932baed419c43390595b5e5cece12/gen/suites/actor_creation/params.go#L41-L65) and [`fails-unparsable-init-actor-exec-msg`](https://github.com/filecoin-project/test-vectors/blob/master/gen/suites/actor_creation/params.go#L15-L39) as the messages in these tests are being applied directly by the `vm`, not being sent through `Runtime.Send`.

The `SysErrSenderInvalid` error is coming from: https://github.com/filecoin-project/lotus/blob/de654de91161b3d71d52b3d9f79679a5dede0d7f/chain/vm/invoker.go#L144

I can confirm that if I change it to `exitcode.ErrSerialization`, that's what we receive in those tests.

@anorth 🙏 can you confirm if `ErrSerialization` is the right code to return in _this_ case and not `SysErrSenderInvalid`? Note that this is on the receiver side, **de**serializing parameters. If so I'll send a PR to lotus to get it fixed.

**Comment by anorth** [comment]
Yes, I confirm that `ErrSerialization` is the right code to return for every case of failure to encode/decode parameters.
